File: Codeforces/Golang/900/96A - Football/main.go
package main

import (
	"fmt"
)

func main() {
	var players string
	fmt.Scan(&players)

	if len(players) < 7 {
		fmt.Println("NO")
		return
	}
	count := 1
	dangerous := false
	for i := 1; i < len(players); i++ {
		if players[i] == players[i-1] {
			count++
			if count == 7 {
				dangerous = true
				break
			}
		} else {
			count = 1
		}
	}

	if dangerous {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}

File: Codeforces/Rust/900/96A - Football/src/main.rs
fn main() {
    let mut player = String::new();
    let _ = std::io::stdin().read_line(&mut player);

    let players = player.trim().as_bytes();
    if players.len() < 7 {
        println!("NO");
        return;
    }
    let mut count = 1;
    let mut dangerous = false;

    for i in 1..players.len() {
        if players[i] == players[i - 1] {
            count += 1;
            if count == 7 {
                dangerous = true;
                break;
            }
        } else {
            count = 1;
        }
    }

    if dangerous {
        println!("YES")
    } else {
        println!("NO")
    }
}
